test/test_time.cpp

```diff
@@ -14,56 +14,64 @@
 
 #include <gtest/gtest.h>
 
-#include <inttypes.h>
-
 #include <chrono>
+#include <cinttypes>
 #include <thread>
 
+#include "osrf_testing_tools_cpp/memory_tools/memory_tools.hpp"
+
 #include "rcutils/error_handling.h"
 #include "rcutils/time.h"
 
-#include "./memory_tools/memory_tools.hpp"
+using osrf_testing_tools_cpp::memory_tools::disable_monitoring_in_all_threads;
+using osrf_testing_tools_cpp::memory_tools::enable_monitoring_in_all_threads;
+using osrf_testing_tools_cpp::memory_tools::expect_no_calloc_begin;
+using osrf_testing_tools_cpp::memory_tools::expect_no_calloc_end;
+using osrf_testing_tools_cpp::memory_tools::expect_no_free_begin;
+using osrf_testing_tools_cpp::memory_tools::expect_no_free_end;
+using osrf_testing_tools_cpp::memory_tools::expect_no_malloc_begin;
+using osrf_testing_tools_cpp::memory_tools::expect_no_malloc_end;
+using osrf_testing_tools_cpp::memory_tools::expect_no_realloc_begin;
+using osrf_testing_tools_cpp::memory_tools::expect_no_realloc_end;
+using osrf_testing_tools_cpp::memory_tools::MemoryToolsService;
+using osrf_testing_tools_cpp::memory_tools::on_unexpected_calloc;
+using osrf_testing_tools_cpp::memory_tools::on_unexpected_free;
+using osrf_testing_tools_cpp::memory_tools::on_unexpected_malloc;
+using osrf_testing_tools_cpp::memory_tools::on_unexpected_realloc;
 
 class TestTimeFixture : public ::testing::Test
 {
 public:
   void SetUp()
   {
-    set_on_unexpected_malloc_callback([]() {ASSERT_FALSE(true) << "UNEXPECTED MALLOC";});
-    set_on_unexpected_realloc_callback([]() {ASSERT_FALSE(true) << "UNEXPECTED REALLOC";});
-    set_on_unexpected_free_callback([]() {ASSERT_FALSE(true) << "UNEXPECTED FREE";});
-    start_memory_checking();
+    osrf_testing_tools_cpp::memory_tools::initialize();
+    on_unexpected_malloc([]() {FAIL() << "UNEXPECTED MALLOC";});
```

Comment:
I know you probably had a reason for favouring the new name, but I did want to point out from a user perspective that `set_*_callback` seemed to better capture what this function was doing. Given the other macros e.g. `EXPECT_NO_MEMORY_OPERATIONS` taking in a set of statements to check, now when I look at something called `on_unexpected_malloc` I expect it to similarly be checking for unexpected malloc on a particular set of statements (which is not what it's doing). We can survive with the naming either way though, for sure.

Reply:
I was following the `on_<event>(callback)` style of things like JQuery, e.g. https://api.jquery.com/on/. Personally I find the new syntax easier to read, but that's subjective. I don't really follow:

> I expect it to similarly be checking for unexpected malloc on a particular set of statements (which is not what it's doing).

Reading that, it's like `on_unexpected_malloc(<statements>)`  becomes "on an unexpected malloc for statements `<statements>` do...?". It's an incomplete thought.

So I agree that `set_*_callback` is very unambiguous, but I don't exactly see the issue with `on_*`, and it's 9 characters shorter.

My personal preference would be to keep the new name, but I'm open to other arguments.

Reply:
Yeah, I don't mean that the function as it's named is exceptionally misleading, just that on first glance I expected that line to be doing _checking_ for unexpected mallocs as opposed to setting a callback, whereas with the other naming it was clearer. If that's a naming pattern used by other languages e.g. jquery as you mentioned, then that's fine now that it's been changed

Reply:
> on first glance I expected that line to be doing checking for unexpected mallocs as opposed to setting a callback

That's what I was talking about too, but the keyword `on` doesn't make sense to me without a callback, as-in `on <something> do <this (callback)>`, so I could imagine `on` taking statements _and_ a callback, but not just statements. So maybe it's important to understand why `on` indicates to you (and potentially users) that something is being checked right there.

At any rate, it's documented:

https://github.com/osrf/osrf_testing_tools_cpp/blob/67b321750a863cd0e2a49a8ddd5b1227031513de/osrf_testing_tools_cpp/include/osrf_testing_tools_cpp/memory_tools/testing_helpers.hpp#L26